Code/GraphMol/FileParsers/ForwardSDMolSupplier.cpp

```diff
@@ -34,6 +34,12 @@ std::string strip(const std::string &orig) {
   res = boost::trim_left_copy_if(res, boost::is_any_of(" \t\r\n"));
   return res;
 }
+std::string_view strip(const std::string_view orig) {
+  std::string_view res = orig;
+  res.remove_prefix(std::min(res.find_first_not_of(" \t\r\n"), res.size()));
+  res.remove_suffix(std::min(res.find_first_not_of(" \t\r\n"), res.size()));
```

Comment:
This doesn't right strip.  I'm getting

```
*foo*
*foo *
*foo *
*foo foo *
```

for my test cases

I think what you want is:

```
res.remove_prefix(std::min(res.find_first_not_of(" \t\r\n"), res.size()));                                                                                                
res.remove_suffix(res.size() - res.find_last_not_of(" \t\r\n") - 1); 
```

Reply:
right you are... thanks for catching that